commit 620ea803d2df7ad5681f0f95be4a5babdee1dc64 (refs/changes/88/2788/1)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2017-10-16T18:30:54+00:00 (2 years ago)
    
    joyent/node-manta#332 Add support for Content-Encoding for objects

diff --git a/bin/mmpu b/bin/mmpu
index 306e00a..60ddc8a 100755
--- a/bin/mmpu
+++ b/bin/mmpu
@@ -154,7 +154,8 @@ MMpu.prototype.do_create = function do_create(subcmd, opts, args, cb) {
         account: opts.account,
         md5: opts.md5,
         copies: opts.copies,
-        size: opts.size
+        size: opts.size,
+        encoding: opts.encoding
     };
 
     client.createUpload(objectPath, createUploadOpts, function (err, obj) {
diff --git a/lib/client.js b/lib/client.js
index ba0495a..36d93f9 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -3405,6 +3405,7 @@ MantaClient.prototype.createUpload = function createUpload(p, opts, cb) {
     assert.optionalNumber(opts.copies, 'opts.copies');
     assert.optionalNumber(opts.size, 'opts.size');
     assert.optionalString(opts.md5, 'opts.md5');
+    assert.optionalString(opts.encoding, 'opts.encoding');
     assert.func(cb, 'callback');
 
     var self = this;
@@ -3454,6 +3455,10 @@ MantaClient.prototype.createUpload = function createUpload(p, opts, cb) {
             targetObjectHeaders['content-md5'] = opts.md5;
         }
 
+        if (opts.encoding) {
+            targetObjectHeaders['content-encoding'] = opts.encoding;
+        }
+
         var log = self.log.child({
             path: url,
             req_id: options.id
